src/Transport/Sending/IBatcher.cs

```diff
@@ -1,10 +1,11 @@
 namespace NServiceBus.Transport.AzureServiceBus
 {
     using System.Collections.Generic;
+    using System.Threading.Tasks;
     using Transport;
 
     interface IBatcherInternal
     {
-        IList<BatchInternal> ToBatches(TransportOperations operations);
+        Task<List<BatchInternal>> ToBatches(TransportOperations operations);
```

Comment:
I think there's some misunderstanding here @yvesgoeleven.
The IO is **not** performed on every dispatch. The result is [cached and re-used](https://github.com/Particular/NServiceBus.AzureServiceBus/blob/d4f6f967bc665d9330e55b753228c369e1d0877a/src/Transport/Topology/Topologies/ForwardingTopologySectionManager.cs#L37).